lxd/instance/drivers/driver_qemu.go

```diff
@@ -2085,6 +2085,8 @@ Documentation=https://linuxcontainers.org/lxd
 ConditionPathExists=/dev/virtio-ports/org.linuxcontainers.lxd
 Before=cloud-init.target cloud-init.service cloud-init-local.service
 DefaultDependencies=no
+After=run-lxd_agent.mount
+Requires=run-lxd_agent.mount
```

Comment:
How does that work though? `run-lxd_agent.mount` shows up as a result of the `ExectStartPre` on the unit.

So making it `Requires=run-lxd_agent.mount` should prevent the unit from ever starting.

Reply:
Apparently systemd is smart enough to handle this gracefully. You can test it with a bogus `.mount` like this:

```
# systemctl edit lxd-agent
[Unit]
After=22run-lxd_agent.mount
Requires=22run-lxd_agent.mount
```

This does not prevent the lxd-agent from starting up.

Reply:
Confirmed that this somehow works, it's a bit odd though.
I was expecting the `After` to do the right thing, `Requires` however I would have expected to be a stronger requirement.